libnavigation-base/src/main/java/com/mapbox/navigation/base/utils/route/RouteUtils.kt

```diff
@@ -0,0 +1,162 @@
+package com.mapbox.navigation.base.utils.route
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.RouteLeg
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.geojson.Point
+import com.mapbox.geojson.utils.PolylineUtils
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.base.utils.DecodeUtils.precision
+import com.mapbox.navigation.utils.internal.logE
+import com.mapbox.navigation.utils.internal.logW
+
+/**
+ * Route Utils
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+object RouteUtils {
+
+    private const val LOG_CATEGORY = "RouteUtils"
+
+    /**
+     * The method checks whenever the route has unexpected closures based
+     * on [RouteOptions.snappingIncludeClosuresList] and [RouteOptions.snappingIncludeStaticClosuresList]
+     * flags. If the any of them is **false** for the waypoint and closure overlaps it the method
+     * immediately returns **true**.
+     *
+     * If route has closures that are not overlapped waypoints, the method immediately returns
+     * **true**.
+     *
+     * Note: Experimental preview API
+     */
+    @ExperimentalPreviewMapboxNavigationAPI
+    fun hasRouteUnexpectedClosures(navigationRoute: NavigationRoute): Boolean {
```

Comment:
```suggestion
    fun NavigationRoute.hasUnexpectedClosures(): Boolean {
```
and move to a public `NavigationRouteEx` file (with JVM file name `NavigationRouteUtils`).

It will also need to be a suspend/callback function (or be instructed to be used from a worker thread) because parsing geometries can be expensive for long routes.